### ISSUE TITLE
Producer and Consumer should opt-out of Sync

### ISSUE THREAD

**Issue by reem** [polyfractal/bounded-spsc-queue#1]
Right now Producer and Consumer are Sync, which allows the user to call `try_{push, pop}` from multiple threads at once, thus violating the single-consumer-single-producer property and possibly causing memory unsafety.

cc @Gankro didn't see anything else obvious

**Comment by polyfractal** [comment]
Thanks!  I'm assuming opting out is basically just:

```rust
impl<T: Send> !Sync for Consumer<T> {}
impl<T: Send> !Sync for Producer<T> {}
```

?

**Comment by reem** [comment]
Yes, though you may want to opt out for unbounded T.

**Comment by polyfractal** [comment]
Fixed in 70b3508070adfbe7a8408c6950f4c7c69be1e994.  Thanks!
